app/models/validations/household_validations.rb

```diff
@@ -20,6 +30,12 @@ def validate_reason_for_leaving_last_settled_home(record)
       record.errors.add :referral, I18n.t("validations.household.referral.reason_permanently_decanted")
       record.errors.add :reason, I18n.t("validations.household.reason.not_internal_transfer")
     end
+
+    return unless record.form.start_year_after_2024?
+
+    if record.reason == 20 && Regexp.union(PHRASES_INDICATING_HOMELESSNESS.map { |phrase| Regexp.new("\\A[^[:alpha:]]*#{phrase}[^[:alpha:]]*\\Z", Regexp::IGNORECASE) }).match?(record.reasonother)
```

Comment:
Would this do the same thing? `PHRASES_INDICATING_HOMELESSNESS.any? { |phrase| Regexp.new("\\A[^[:alpha:]]*#{phrase}[^[:alpha:]]*\\Z", Regexp::IGNORECASE).match?(record.reason)}`

Any way we could do it without regex? Idm if not

Reply:
Don't think we can avoid the regex. I've moved the union so that we're only doing it once, which is definitely the less-work-to-run option and hopefully reads a bit more nicely - lmk what you think.